compiler/qsc_project/src/fs.rs

```diff
@@ -13,13 +16,46 @@ use std::{path::PathBuf, sync::Arc};
 #[derive(Default)]
 pub struct StdFs;
 
+impl DirEntry for PathBuf {
+    type Error = Infallible;
+
+    fn entry_type(&self) -> Result<EntryType, Self::Error> {
+        if self.is_file() {
+            Ok(EntryType::File)
+        } else if self.is_dir() {
+            Ok(EntryType::Folder)
+        } else if self.is_symlink() {
+            Ok(EntryType::Symlink)
+        } else {
+            unreachable!()
+        }
+    }
+
+    fn entry_extension(&self) -> String {
+        self.extension()
+            .map(|x| x.to_string_lossy().to_string())
+            .unwrap_or_default()
+    }
+
+    fn entry_name(&self) -> String {
+        self.file_name()
+            .expect("canonicalized symlink cannot end in relative path")
+            .to_string_lossy()
+            .to_string()
+    }
+
```

Comment:
Do we actually need `entry_extension` and `entry_name` on the trait? Looks like all three implementations do the same thing - call the corresponding method on `PathBuf`. The caller can just retrieve the `PathBuf` and do this work itself, instead of us having 3 duplicate implementations.